test/index.test.js

```diff
@@ -693,6 +693,26 @@ describe('mongoose module:', function() {
     assert.ok(mongoose.isValidObjectId(new mongoose.Types.ObjectId()));
     assert.ok(!mongoose.isValidObjectId(6));
   });
+  it('allows global `strictPopulate` option (gh-10694)', async function() {
```

Comment:
Splitting each test case into a different `it(...)` invocation makes it easier to understand the intention.

I'd split the test here into the following different test cases:
```js
it('global `strictPopulate` works when false')
it('`global `strictPopulate` works when true');
it('global `strictPopulate` can be overriden on a specific query set to false')
it('global `strictPopulate` can be overriden on a specific query set to true')
```

And I'd do my best to make the test case look like a real-world scenario by avoiding variable names such as `test` on line 708.